### PR TITLE
CTM-ETM coupling deliverable 3b

### PR DESCRIPTION
**Pull request by kndehaan**
This PR implements the heat burner solution as part of deliverable 3b from the CTM-ETM coupling project. This PR comes with the following features:

- Heat nodes are added for all six industry CHP node
- External coupling inputs are added to set electricity and heat efficiency of the industry CHP
- External coupling inputs are added to set industry CHP electricity output capacity, which will 'activate' the related heat nodes for the heat provision part
- Queries are updated to rectify fallout

### CODE REVIEW

File: graphs/energy/nodes/industry/industry_chp_engine_gas_power_fuelmix.ad [hunk 1]
```diff
@@ -3,6 +3,7 @@
 - use = energetic
 - presentation_group = chps
 - graph_methods = [demand, output, full_load_hours]
+- waste_outputs = [steam_hot_water]
```
Comment (mabijkerk) [inline graphs/energy/nodes/industry/industry_chp_engine_gas_power_fuelmix.ad:6]: In the description of https://github.com/quintel/etsource/commit/139c40182662ea8cd545ab27c4bf8f2629fffa21 it is stated for which nodes the `steam_hot_water` attribute should be added. I think we should check whether it makes sense to add them to all CHPs now.

File: graphs/energy/nodes/industry/industry_chp_turbine_gas_power_fuelmix.ad [hunk 2]
```diff
@@ -3,6 +3,7 @@
 - use = energetic
 - presentation_group = chps
 - graph_methods = [demand, output, full_load_hours]
+- waste_outputs = [steam_hot_water]
```
Comment (mabijkerk) [inline graphs/energy/nodes/industry/industry_chp_turbine_gas_power_fuelmix.ad:6]: Similar comment, see https://github.com/quintel/etsource/commit/139c40182662ea8cd545ab27c4bf8f2629fffa21

File: graphs/energy/nodes/industry/industry_chp_turbine_hydrogen.ad [hunk 3]
```diff
@@ -5,6 +5,7 @@
 - use = energetic
 - presentation_group = chps
 - graph_methods = [demand, output, full_load_hours]
+- waste_outputs = [steam_hot_water]
```
Comment (mabijkerk) [inline graphs/energy/nodes/industry/industry_chp_turbine_hydrogen.ad:8]: Similar comment, see https://github.com/quintel/etsource/commit/139c40182662ea8cd545ab27c4bf8f2629fffa21

File: graphs/energy/nodes/industry/industry_chp_ultra_supercritical_coal.ad [hunk 4]
```diff
@@ -18,25 +18,13 @@
...
-- hours_maint_nl = 32040.0
-- hours_place_nl = 193680.0
-- hours_prep_nl = 26820.0
-- hours_prod_nl = 0.0
-- hours_remov_nl = 21600.0
 
 ~ demand = CENTRAL_PRODUCTION(industry_chp_ultra_supercritical_coal, demand)
 ~ full_load_hours = CENTRAL_PRODUCTION(industry_chp_ultra_supercritical_coal, full_load_hours)
-~ input.coal = 1.0
-~ input.torrefied_biomass_pellets = 0.0
```
Comment (mabijkerk) [inline graphs/energy/nodes/industry/industry_chp_ultra_supercritical_coal.ad:40]: Nice catch!

File: graphs/energy/nodes/industry/industry_heat_engine_gas_power_fuelmix.ad [hunk 5]
```diff
@@ -0,0 +1,31 @@
+# this node is activated once input for external_coupling_capacity_of_industry_chp_combined_cycle_gas_power_fuelmix is set
```
Comment (mabijkerk) [inline graphs/energy/nodes/industry/industry_heat_engine_gas_power_fuelmix.ad:1]: The descriptions of these nodes do not always match.

File: graphs/energy/nodes/industry/industry_heat_engine_gas_power_fuelmix.ad [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+# this node provides the heat output of the connected chp
+# full_load_hours are an estimation for the must-run characteristic of heat supply from industrial CHPs
+# steam hot water efficiency is a generic value which will be updated when external coupling is activated
+# cost attributes are set to 0 since these are allocated to the connected CHP
+
+- output.loss = elastic
+- output.steam_hot_water = 0.9
+- groups = [preset_demand, heat_production, wacc_proven_tech, costs_production_heat_plants, sankey_conversion_group]
+- use = energetic
+- presentation_group = traditional_heat
```
Comment (mabijkerk) [inline graphs/energy/nodes/industry/industry_heat_engine_gas_power_fuelmix.ad:11]: Note that this attribute is only relevant when the technical & financial properties of the node are shown in the frontend through the `related_node` attribute on ETModel.

**Comment by mabijkerk** [comment]
@kndehaan I think we should set the priority of the 'regular' inputs for heat and electricity efficiencies to 1 as well. This way, if you have not set the external coupling efficiencies, but you have set the 'regular' efficiencies, you will still get the same results. Do you agree? If so, we should check that it does not have any fallout for the 'regular' capacity inputs.

This relates to the meta discussion about input sets: do you always set all the external coupling inputs (in this case capacity, electricity and heat efficiency) or can set individual external coupling inputs (for example only electricity efficiency) and still expect the coupling to work?

File: gqueries/general/conversion/heat/network_gas_in_source_of_central_heat_production.gql [hunk 7]
```diff
@@ -1,13 +1,16 @@
...
+        SUM(
+          V(
+            Q(central_heat_producing_converters_sankey),
+            "output_of_steam_hot_water * network_gas_input_conversion"
+          )
+        ),
+        BILLIONS
       ),
-      BILLIONS
+      Q(gas_power_fuelmix_in_source_of_central_heat_conversion) * V(energy_mixer_for_gas_power_fuel, network_gas_input_conversion)
```
Comment (mabijkerk) [inline gqueries/general/conversion/heat/network_gas_in_source_of_central_heat_production.gql:14]: If we're including the `network_gas` share in the `gas_power_fuelmix` here, shouldn't we similarly include the `crude_oil` share in the `oil_and_derivatives_in_source_of_central_heat_production` and the `bio_oil` share in `biomass_products_in_source_of_central_heat_production`?

File: gqueries/output_elements/output_series/table_141_chp_properties/industry_chp_combined_cycle_gas_power_fuelmix_heat_production_in_chp_properties.gql [hunk 8]
```diff
@@ -1,4 +1,15 @@
 # heat production of industry_chp_combined_cycle_gas_power_fuelmix
+# accounts for CHP heat component (industry_heat_xxx) when external coupling is activated
 
 - unit = PJ
-- query = future:UNIT(V(industry_chp_combined_cycle_gas_power_fuelmix, "output_of_useable_heat + output_of_steam_hot_water"),billions)
+- query =
+    future:UNIT(
```
Comment (mabijkerk) [inline gqueries/output_elements/output_series/table_141_chp_properties/industry_chp_combined_cycle_gas_power_fuelmix_heat_production_in_chp_properties.gql:6]: I'm not a big fan of the function `UNIT`, can't we simply use the `DIVIDE` and `BILLIONS` functions? These seem to be used more generally through the model.

**Comment by kndehaan** [comment]
> @kndehaan I think we should set the priority of the 'regular' inputs for heat and electricity efficiencies to 1 as well. This way, if you have not set the external coupling efficiencies, but you have set the 'regular' efficiencies, you will still get the same results. Do you agree? If so, we should check that it does not have any fallout for the 'regular' capacity inputs.

Good point, I agree. I will change the priorities of the regular inputs for heat and electricity efficiencies and test whether this still works as expected.

File: gqueries/output_elements/output_series/table_141_chp_properties/industry_chp_combined_cycle_gas_power_fuelmix_heat_production_in_chp_properties.gql (hunk 8 above)
Comment (kndehaan) [reply gqueries/output_elements/output_series/table_141_chp_properties/industry_chp_combined_cycle_gas_power_fuelmix_heat_production_in_chp_properties.gql:6]: I've now applied `DIVIDE` and `BILLIONS` to the CHP queries related to this project.

File: gqueries/general/conversion/heat/network_gas_in_source_of_central_heat_production.gql (hunk 7 above)
Comment (kndehaan) [reply gqueries/general/conversion/heat/network_gas_in_source_of_central_heat_production.gql:14]: Good point, I've now added gas power fuelmix to the oil and biomass products queries.

File: graphs/energy/nodes/industry/industry_chp_engine_gas_power_fuelmix.ad (hunk 1 above)
Comment (mabijkerk) [reply graphs/energy/nodes/industry/industry_chp_engine_gas_power_fuelmix.ad:6]: See [docs](https://docs.energytransitionmodel.com/main/merit-order#marginal-costs-of-chps) for the reasoning here..

**Pull request merged**